External/Plugins/HaXeContext/Context.cs

```diff
@@ -1381,10 +1384,66 @@ internal void OnFunctionCompletionResult(HaxeComplete hc, HaxeCompleteStatus sta
 
                 case HaxeCompleteStatus.TYPE:
                     hc.Expr.Position--;
-                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, hc.Type, null, true);
+                    ASComplete.FunctionContextResolved(hc.Sci, hc.Expr, result.Type, null, true);
                     break;
             }
         }
+
+        public override bool HandleGotoDeclaration(ScintillaControl sci, ASExpr expression)
+        {
+            var hc = new HaxeComplete(sci, expression, false, completionModeHandler, HaxeCompleteType.POSITION);
+            hc.GetPosition(OnPositionCompletionResult);
+            return true;
+        }
+
+        internal void OnPositionCompletionResult(HaxeComplete hc, HaxePositionCompleteResult result, HaxeCompleteStatus status)
+        {
+            switch (status)
+            {
+                case HaxeCompleteStatus.ERROR:
+                    TraceManager.AddAsync(hc.Errors, -3);
+                    break;
+
+                case HaxeCompleteStatus.POSITION:
+                    ASComplete.SaveLastLookupPosition(hc.Sci);
+
+                    PluginBase.MainForm.OpenEditableDocument(result.Path, false);
+                    ScintillaControl sci = PluginBase.MainForm.CurrentDocument.SciControl;
+
+                    if (sci.InvokeRequired)
+                    {
+                        sci.BeginInvoke((MethodInvoker)delegate
+                        {
+                            HighlightDeclaration(sci, result);
+                        });
+                    }
+                    else HighlightDeclaration(sci, result);
+                    break;
+            }
+        }
+
+        internal void HighlightDeclaration(ScintillaControl sci, HaxePositionCompleteResult result)
```

Comment:
Yeah, this would be fixed if this was made consistent with FD's current goto declaration highlighting behavior like I mentioned, using `ASComplete.LocateMember()`. The field name is necessary for that.